vm.h

```diff
@@ -31,6 +32,19 @@ typedef struct {
     vm_handler handler;
 } vm_opcode_impl;
 
+typedef struct {
+    enum { SYMBOLIC, NUMERIC } type;
+    char *label;
+    int next_pc;
+} vm_label;
+
+typedef enum {
+    LABEL_NOT_FOUND = -1,
+    UNIMPLEMNET_ERR1,
```

Comment:
Leaving one `UNIMPLMENTED` should be enough. You can rewrite parts of file `opcode.c` manipulations.